miner/src/miner.rs

```diff
@@ -132,7 +132,20 @@ impl MinerService for Miner {
 			self.extra_data(),
 			transactions,
 		);
-		*self.sealing_block.lock().unwrap() = b;
+
+		match b {
```

Comment:
True - in this case it looks much better. However in general I'm trying to avoid side effects in `map`.